phoenix-core-server/src/main/java/org/apache/phoenix/replication/ReplicationLogGroup.java

```diff
@@ -0,0 +1,352 @@
+         * This is the default and primary mode of operation.
+         */
+        SYNC,
+
+        /**
+         * Fallback mode when the standby cluster's HDFS is unavailable. Mutations are stored
+         * locally and will be forwarded when connectivity is restored.
+         */
+        STORE_AND_FORWARD,
+
+        /**
+         * Transitional mode where new mutations are written directly to the standby cluster
+         * while concurrently draining the local queue of previously stored mutations. This mode
+         * is entered when connectivity to the standby cluster is restored while there are still
+         * mutations in the local queue.
+         */
+        SYNC_AND_FORWARD;
+    }
+
+    /**
+     * Get or create a ReplicationLogGroup instance for the given HA Group.
+     *
+     * @param conf Configuration object
+     * @param serverName The server name
+     * @param haGroupId The HA Group identifier
+     * @return ReplicationLogGroup instance
+     * @throws RuntimeException if initialization fails
+     */
+    public static ReplicationLogGroup get(Configuration conf, ServerName serverName,
+            String haGroupId) {
+        return INSTANCES.computeIfAbsent(haGroupId, k -> {
+            try {
+                ReplicationLogGroup group = new ReplicationLogGroup(conf, serverName, haGroupId);
+                group.init();
+                return group;
+            } catch (IOException e) {
+                LOG.error("Failed to create ReplicationLogGroup for HA Group: {}", haGroupId, e);
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    /**
+     * Protected constructor for ReplicationLogGroup.
+     *
+     * @param conf Configuration object
+     * @param serverName The server name
+     * @param haGroupId The HA Group identifier
+     */
+    protected ReplicationLogGroup(Configuration conf, ServerName serverName, String haGroupId) {
+        this.conf = conf;
+        this.serverName = serverName;
+        this.haGroupId = haGroupId;
+    }
+
+    /**
+     * Initialize the ReplicationLogGroup by creating the appropriate writer implementation.
+     *
+     * @throws IOException if initialization fails
+     */
+    protected void init() throws IOException {
+        // For now, we initialize only the remote writer and set the mode to SYNC.
+        remoteWriter = new StandbyLogGroupWriter(conf, serverName, haGroupId);
+        remoteWriter.init();
+        mode = ReplicationMode.SYNC;
+        // TODO: Initialize the local writer (StoreAndForwardLogGroupWriter).
+        // TODO: Switch the initial mode to STORE_AND_FORWARD if the remote writer fails to
+        // initialize.
+        localWriter = null;
+        LOG.info("Started ReplicationLogGroup for HA Group: {}", haGroupId);
+    }
+
+    /**
+     * Get the HA Group ID managed by this instance.
+     *
+     * @return HA Group ID
+     */
+    public String getHaGroupId() {
```

Comment:
I think we need to use Name instead of ID. ID implies that an HA group has also a name. The ID also implies that it is generated instead of given. An HA group has a name and the name is given by the user. The HA feature is not responsible for the uniqueness of HA group names among Phoenix clusters.

Reply:
Addressed in 943239154